### PR TITLE
#536: Delete cluster init buckets after cluster is deleted

### PR DESCRIPTION
**Pull request by MatthewBemis**
Cleanup script to come.

Have you read [CONTRIBUTING.md](../CONTRIBUTING.md) lately? If not, do that first.

I, the developer opening this PR, do solemnly pinky swear that:

- [x] I've documented my API changes in Swagger

In all cases:

- [x] Get two thumbsworth of review and PO signoff if necessary
- [x] Verify all tests go green
- [ ] Squash and merge; you can delete your branch after this
- [ ] Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

File: src/main/scala/org/broadinstitute/dsde/workbench/leonardo/monitor/ClusterMonitorActor.scala [hunk 1]
```diff
@@ -229,6 +229,9 @@ class ClusterMonitorActor(val cluster: Cluster,
     logger.info(s"Cluster ${cluster.projectNameString} has been deleted.")
 
     for {
+      // delete the init bucket so we don't continue to accrue costs after cluster is deleted
+      _ <- deleteInitBucket
```
Comment (MatthewBemis) [inline src/main/scala/org/broadinstitute/dsde/workbench/leonardo/monitor/ClusterMonitorActor.scala]: how convenient that this code was left behind 😄

**Comment by MatthewBemis** [comment]
jenkins retest because there were some build failures that appear to be out of our control (external deps not downloading)

File: src/main/scala/org/broadinstitute/dsde/workbench/leonardo/monitor/ClusterMonitorActor.scala (hunk 1 above)
Comment (rtitle) [reply src/main/scala/org/broadinstitute/dsde/workbench/leonardo/monitor/ClusterMonitorActor.scala]: Heh, yeah it was a complete oversight that we missed deleting this bucket at cluster deletion time

**Review by rtitle — APPROVED** [review]
:shipit:

**Review by kyuksel — APPROVED** [review]
👍 Bernick will be happy.

**Pull request merged**